Add Testcontainers Support for Postgres, MariaDB, MySQL

javahippie opened the pull request:
This pull request should provide the first Testcontainers abilities to the project to build upon.

They can be run locally with: `./mvnw -Pdistro,mariadb install` or `./mvnw -Pdistro,postgresql install` or `./mvnw -Pdistro,mysql install`

These builds are running successfully for me.

Fixes #549
Fixes #550 
Fixes #551 
Fixes #2
Related to #548

kthoms reviewed (changes requested):
Can't test it locally. In my Rancher environment it does not build.

Please add "mariadb" to Integration-tests.yml and start an integration test from your branch.

javahippie commented on test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonMariaDBContainerProvider.java:
```diff
@@ -34,6 +34,6 @@ public boolean supports(String databaseType) {
   public JdbcDatabaseContainer newInstance(String tag) {
```
The overriding is avoiding collisions on the default containers, that's also why the TC string contains those identifiers. I like it better than just overriding the default, tbh. The values obviously will be changed

javahippie commented:
Integration Test with Postgres started here: https://github.com/operaton/operaton/actions/runs/13591301372/job/37997853842

javahippie commented:
On my local machine, the latest changes managed to run Postgres on Tomcat with Testcontainers:

`./mvnw -Pengine-integration,tomcat,postgresql verify -f qa`

```
[INFO] Reactor Summary for Operaton - QA 1.0.0-beta-4-SNAPSHOT:
[INFO] 
[INFO] Operaton - QA ...................................... SUCCESS [  0.147 s]
[INFO] Operaton - QA - Ensure clean DB plugin ............. SUCCESS [  0.432 s]
[INFO] Operaton - QA Tomcat Test Runtime .................. SUCCESS [  1.698 s]
[INFO] Operaton - QA Integration Tests - Engine ........... SUCCESS [36:37 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  36:39 min
[INFO] Finished at: 2025-03-02T01:08:01+01:00
[INFO] ------------------------------------------------------------------------
```

GitHub Actions are not yet working, might be an issue with the Docker Daemon there. Still looking into it

javahippie commented:
@kthoms I addressed your review comments and the build was verified by Antonie (see Slack). Are there still changes you would like to see? I think we need a different pipeline/job because the integration-tests.yml is currently focussed on the qa folder and Arquillian, but I'd see this as a separate ticket

kthoms commented on qa/tomcat-runtime/pom.xml:
```diff
@@ -40,6 +40,52 @@
       <scope>runtime</scope>
     </dependency>
 
+    <dependency>
```
Could you use the Testcontainers BOM
```
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.testcontainers</groupId>
        <artifactId>testcontainers-bom</artifactId>
        <version>${version.testcontainers}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
```
Then leave our `<version>` tags here.

Also update the version in parent/pom.xml

javahippie replied:
Good catch!

kthoms reviewed (approved):
I hope it is OK that I made the dependency management change

javahippie commented:
@kthoms completely fine, thanks! Having a busy day, would not have found the time today.
